dev/com.ibm.ws.messaging.open_jms20deliverydelay_fat/fat/src/com/ibm/ws/messaging/jms20/deliverydelay/fat/DelayFullTest.java

```diff
@@ -0,0 +1,584 @@
+    }
+
+    private boolean runInServlet(String test) throws IOException {
+        return TestUtils.runInServlet(clientHost, clientPort, ddContextRoot, test); // throws IOException
+    }
+
+    @Test
+    public void testDeliveryDelayForDifferentDelays_B() throws Exception {
+        restartClient(MDB_CONFIG_QUEUE_BINDINGS);
+
+        runInServlet("testDeliveryDelayForDifferentDelays");
+
+        String msg = clientServer.waitForStringInLogUsingLastOffset("Message received on mdb : QueueBindingsMessage2");
+        assertNotNull("Could not find the upload message in the trace.log", msg);
+        msg = clientServer.waitForStringInLogUsingLastOffset("Message received on mdb : QueueBindingsMessage1");
+        assertNotNull("Could not find the upload message in the trace.log", msg);
+
+        restartServers(MDB_CONFIG_QUEUE_TCP);
+
+        runInServlet("testDeliveryDelayForDifferentDelays_Tcp");
+
+        msg = clientServer.waitForStringInLogUsingLastOffset("Message received on mdb : QueueTCPMessage2");
+        assertNotNull("Could not find the upload message in the trace.log", msg);
+        msg = clientServer.waitForStringInLogUsingLastOffset("Message received on mdb : QueueTCPMessage1");
+        assertNotNull("Could not find the upload message in the trace.log", msg);
+
+        restartServers();
+    }
+
+    @Test
+    public void testDeliveryDelayForDifferentDelaysTopic_B() throws Exception {
+        restartClient(MDB_CONFIG_TOPIC_BINDINGS);
+
+        runInServlet("testDeliveryDelayForDifferentDelaysTopic");
+
+        String msg = clientServer.waitForStringInLogUsingLastOffset("Message received on mdb : TopicBindingsMessage2");
+        assertNotNull("Could not find the upload message in the trace.log", msg);
+        msg = clientServer.waitForStringInLogUsingLastOffset("Message received on mdb : TopicBindingsMessage1");
+        assertNotNull("Could not find the upload message in the trace.log", msg);
+
+        restartServers(MDB_CONFIG_TOPIC_TCP);
+
+        runInServlet("testDeliveryDelayForDifferentDelaysTopic_Tcp");
+
+        msg = clientServer.waitForStringInLogUsingLastOffset("Message received on mdb : TopicTCPMessage2");
+        assertNotNull("Could not find the upload message in the trace.log", msg);
+        msg = clientServer.waitForStringInLogUsingLastOffset("Message received on mdb : TopicTCPMessage1");
+        assertNotNull("Could not find the upload message in the trace.log", msg);
+
+        restartServers();
+    }
+
+    @Test
+    public void testPersistentMessageStore_B() throws Exception {
+        // dz restartServers(DEFAULT_CLIENT);
+
+        runInServlet("testPersistentMessage");
+
+        restartServers();
+
+        boolean testResult = runInServlet("testPersistentMessageReceive");
+        assertTrue("testPersistentMessageStore_B failed", testResult);
+    }
+
+    @Test
+    public void testPersistentMessageStore_Tcp() throws Exception {
+        // dz restartServers(DEFAULT_CLIENT);
```

Comment:
We can pull these `// dz` comments.  I added them to debug whether the containing test methods failed due to unexpected client|server state and configuration.  This is a poorly designed test case which relies on the order of test execution, and where tests like this one do not verify (assert) the state and configuration of the messaging client and servers are running before verifying the use-case behavior.

Reply:
Noted; the comments were removed.